integration-tests/tests/utils/config.py

```diff
@@ -77,6 +104,107 @@ def __post_init__(self) -> None:
             )
             raise ValueError(err_msg)
 
+        # Initialize and create required cache directory for package tests.
+        object.__setattr__(self, "temp_config_dir", self.test_root_dir / "temp_config_files")
+        self.test_root_dir.mkdir(parents=True, exist_ok=True)
+        self.temp_config_dir.mkdir(parents=True, exist_ok=True)
+
+        # Write the temporary config file that the package will use.
+        clp_config_obj = self.build_config()
```

Comment:
could be replaced by `get_clp_config_from_mode()` and then you only need to pass in the mode name